src/irr/asset/CImageLoaderJPG.cpp

```diff
@@ -274,20 +291,48 @@ asset::SAssetBundle CImageLoaderJPG::loadAsset(io::IReadFile* _file, const asset
 	
 	// Get image data
 	uint32_t rowspan = cinfo.image_width * cinfo.out_color_components;
-	uint32_t imageSize[3] = {cinfo.image_width,cinfo.image_height,1};
-	uint32_t& width = imageSize[0];
-	uint32_t& height = imageSize[1];
+
+    // OpenGL cannot transfer rows with arbitrary padding
+    static const uint32_t MAX_PITCH_ALIGNMENT = 8u;
+    // try with largest alignment first
+    auto calcPitchInBlocks = [](uint32_t width, uint32_t blockByteSize) -> uint32_t
+    {
+        auto rowByteSize = width * blockByteSize;
+        for (uint32_t _alignment = MAX_PITCH_ALIGNMENT; _alignment > 1u; _alignment >>= 1u)
+        {
+            auto paddedSize = core::alignUp(rowByteSize, _alignment);
+            if (paddedSize % blockByteSize)
+                continue;
+            return paddedSize / blockByteSize;
+        }
+        return width;
+    };
```

Comment:
why is this not factored out to a common static function for some base class like `IImageLoader` and `IImageWriter`

Reply:
should I put it into IImageLoader?